### PR TITLE
Add StatementOptions

### LINKED ISSUE #1419: SQLQuery option for Statement.setFetchSize(int)
Currently setting fetchSize hint for Statement requires custom implementation using SQLDetailedListener. As Statement.setFetchSize(int) is part of standard JDBC API, it would make sense to implement this in core API.

### PR DESCRIPTION
**Pull request by johnktims**
Fixes #1419

### CODE REVIEW

**Comment by johnktims** [comment]
This still needs some tests.

File: querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQuery.java [hunk 1]
```diff
@@ -237,6 +238,17 @@ public void close() throws SQLException {
         }
     }
 
+    private PreparedStatement getPreparedStatement(String queryString) throws SQLException {
+        PreparedStatement statement = conn.prepareStatement(queryString);
+
+        statement.setFetchSize(statementOptions.getFetchSize());
```
Comment (timowest) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQuery.java]: call the setter only when the argument is non null

File: querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQuery.java [hunk 2]
```diff
@@ -67,6 +67,7 @@
     private Object lastCell;
 
     private SQLListenerContext parentContext;
+    private StatementOptions statementOptions = new StatementOptions.Builder().build();
```
Comment (timowest) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQuery.java]: maybe via StatementOptions.builder()...build()

builder() as a factory method for the builder

**Comment by johnktims** [comment]
Thanks for cleaning this up and adding a test. This looks good to me now.

**Comment by Shredder121** [comment]
Yes, this looks good to me as well.

**Pull request merged**